terraform-beyond-layer0/example-2/main.tf

```diff
@@ -0,0 +1,91 @@
+#
+# ./example2/main.tf
+#
+
+provider "layer0" {
+  endpoint        = "${var.endpoint}"
+  token           = "${var.token}"
+  skip_ssl_verify = true
+}
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region     = "${var.region}"
+}
+
+# create one layer0 environment for each map value defined in variable 'environments'
+resource "layer0_environment" "demo" {
+  count = "${length(var.environments)}"
+
+  name = "${lookup(var.environments, count.index)}_demo"
```

Comment:
While that architecture can definitely work, it somewhat undermines the concept of Layer0 environments. 
We've designed them as a shared space for multiple services to run. 

I think the preferred architecture we want to advertise is: 
*  dev
    * application1
    * application2
*  staging
    * application1
    * application2

Admittedly, it doesn't come across very well in this example since we only run one service in the environment. 
Curious to here others' input @diemonster @tlake